cpp/unreal_plugins/WheeledVehicle/Source/WheeledVehicle/VehiclePawn.cpp

```diff
@@ -0,0 +1,177 @@
+
+    FRotator camera_orientation(
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.PITCH"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.YAW"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.ROLL"));
+
+    camera_component_ = CreateDefaultSubobject<UCameraComponent>(TEXT("AVehiclePawn::camera_component_"));
+    SP_ASSERT(camera_component_);
+
+    camera_component_->SetRelativeLocationAndRotation(camera_location, camera_orientation);
+    camera_component_->SetupAttachment(skeletal_mesh_component_);
+    camera_component_->bUsePawnControlRotation = false;
+    camera_component_->FieldOfView = Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.FOV");
+
+    // Setup IMU sensor
+    FVector imu_location(
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.POSITION_X"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.POSITION_Y"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.POSITION_Z"));
+
+    FRotator imu_orientation(
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.PITCH"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.YAW"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.ROLL"));
+
+    imu_component_ = CreateDefaultSubobject<UBoxComponent>(TEXT("AVehiclePawn::imu_component_"));
+    ASSERT(imu_component_);
+
+    imu_component_->SetRelativeLocationAndRotation(imu_location, imu_orientation);
+    imu_component_->SetupAttachment(skeletal_mesh_component_);
+
+    // Setup Sonar sensor
+    FVector sonar_location(
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.SONAR_COMPONENT.POSITION_X"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.SONAR_COMPONENT.POSITION_Y"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.SONAR_COMPONENT.POSITION_Z"));
+
+    FRotator sonar_orientation(
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.SONAR_COMPONENT.PITCH"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.SONAR_COMPONENT.YAW"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.SONAR_COMPONENT.ROLL"));
+
+    sonar_component_ = CreateDefaultSubobject<UBoxComponent>(TEXT("AVehiclePawn::sonar_component_"));
+    ASSERT(sonar_component_);
+
+    sonar_component_->SetRelativeLocationAndRotation(sonar_location, sonar_orientation);
+    sonar_component_->SetupAttachment(skeletal_mesh_component_);
+}
+
+AVehiclePawn::~AVehiclePawn()
+{
+    SP_LOG_CURRENT_FUNCTION();
+}
+
+void AVehiclePawn::SetupPlayerInputComponent(UInputComponent* input_component)
+{
+    SP_ASSERT(input_component);
+    APawn::SetupPlayerInputComponent(input_component);
+
+    input_component->BindAxis("MoveForward", this, &AVehiclePawn::moveForward);
+    input_component->BindAxis("MoveRight", this, &AVehiclePawn::moveRight);
```

Comment:
When enable `WheeledVehicle ` keyboard control, it will invoke`MoveForward` and `MoveRight` every tick sequencially and only torques from `MoveRight` are applied, therefore the pawn could only rotate but never move forward. It can be undesirable when using keyboard control.

Reply:
Agreed with Renhan. This behavior is undesirable. Either fix if you think it is useful for debugging, or remove this keyboard functionality.